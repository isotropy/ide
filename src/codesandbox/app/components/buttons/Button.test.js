import React from 'react';
import testRender from 'codesandbox/app/utils/test/render';
import { MemoryRouter } from 'react-router-dom';
import Button from './Button';

describe('Button', () => {
  it('renders', () => {
    testRender(<Button>Test</Button>);
  });

  it('renders onClick', () => {
    testRender(<Button onClick={() => {}}>Test</Button>);
  });

  it('renders to', () => {
    testRender(
      <MemoryRouter>
        <Button to="https://ivesvh.com">Test</Button>
      </MemoryRouter>
    );
  });

  it('renders href', () => {
    testRender(<Button href="https://ivesvh.com">Test</Button>);
  });

  it('renders properties', () => {
    testRender(<Button small>Test</Button>);
  });

  it('renders disabled', () => {
    testRender(<Button disabled>Test</Button>);
  });

  it('renders secondary', () => {
    testRender(<Button secondary>Test</Button>);
  });
});
