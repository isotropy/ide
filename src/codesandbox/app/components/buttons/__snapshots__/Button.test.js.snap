// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
.c0 {
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  -webkit-animation-name: cUPaoU;
  animation-name: cUPaoU;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  border: none;
  outline: none;
  background-image: linear-gradient(270deg,#fed29d,#A58B66,#7abae8,#56a0d6);
  background-size: 720%;
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 1.125rem;
  text-align: center;
  color: white;
  font-weight: 400;
  box-shadow: 0 3px 3px rgba(0,0,0,0.5);
  width: inherit;
  padding: 0.65rem 2.25rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-animation-name: eEnBCQ;
  animation-name: eEnBCQ;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  box-shadow: 0 7px 10px rgba(0,0,0,0.5);
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c0:active {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
  box-shadow: 0 0 0 rgba(0,0,0,0.5);
}

<button
  className="c0"
>
  Test
</button>
`;

exports[`Button renders disabled 1`] = `
.c0 {
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  -webkit-animation-name: cUPaoU;
  animation-name: cUPaoU;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  border: none;
  outline: none;
  background: rgb(25,29,31);
  background-size: 720%;
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 1.125rem;
  text-align: center;
  color: rgb(69,79,84);
  font-weight: 400;
  width: inherit;
  padding: 0.65rem 2.25rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-decoration: none;
}

<button
  className="c0"
  disabled={true}
>
  Test
</button>
`;

exports[`Button renders href 1`] = `
.c0 {
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  -webkit-animation-name: cUPaoU;
  animation-name: cUPaoU;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  border: none;
  outline: none;
  background-image: linear-gradient(270deg,#fed29d,#A58B66,#7abae8,#56a0d6);
  background-size: 720%;
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 1.125rem;
  text-align: center;
  color: white;
  font-weight: 400;
  box-shadow: 0 3px 3px rgba(0,0,0,0.5);
  width: inherit;
  padding: 0.65rem 2.25rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-animation-name: eEnBCQ;
  animation-name: eEnBCQ;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  box-shadow: 0 7px 10px rgba(0,0,0,0.5);
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c0:active {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
  box-shadow: 0 0 0 rgba(0,0,0,0.5);
}

<a
  className="c0"
  href="https://ivesvh.com"
>
  Test
</a>
`;

exports[`Button renders onClick 1`] = `
.c0 {
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  -webkit-animation-name: cUPaoU;
  animation-name: cUPaoU;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  border: none;
  outline: none;
  background-image: linear-gradient(270deg,#fed29d,#A58B66,#7abae8,#56a0d6);
  background-size: 720%;
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 1.125rem;
  text-align: center;
  color: white;
  font-weight: 400;
  box-shadow: 0 3px 3px rgba(0,0,0,0.5);
  width: inherit;
  padding: 0.65rem 2.25rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-animation-name: eEnBCQ;
  animation-name: eEnBCQ;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  box-shadow: 0 7px 10px rgba(0,0,0,0.5);
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c0:active {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
  box-shadow: 0 0 0 rgba(0,0,0,0.5);
}

<button
  className="c0"
  onClick={[Function]}
>
  Test
</button>
`;

exports[`Button renders properties 1`] = `
.c0 {
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  -webkit-animation-name: cUPaoU;
  animation-name: cUPaoU;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  border: none;
  outline: none;
  background-image: linear-gradient(270deg,#fed29d,#A58B66,#7abae8,#56a0d6);
  background-size: 720%;
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 1.125rem;
  text-align: center;
  color: white;
  font-weight: 400;
  box-shadow: 0 3px 3px rgba(0,0,0,0.5);
  width: inherit;
  padding: 0.5rem 0.75rem;
  font-size: 0.875rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-animation-name: eEnBCQ;
  animation-name: eEnBCQ;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  box-shadow: 0 7px 10px rgba(0,0,0,0.5);
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c0:active {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
  box-shadow: 0 0 0 rgba(0,0,0,0.5);
}

<button
  className="c0"
>
  Test
</button>
`;

exports[`Button renders secondary 1`] = `
.c0 {
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  -webkit-animation-name: cUPaoU;
  animation-name: cUPaoU;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  border: none;
  outline: none;
  background: #3A4B5D;
  background-size: 720%;
  border: 1px solid #56a0d6;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 1.125rem;
  text-align: center;
  color: #56a0d6;
  font-weight: 400;
  box-shadow: 0 3px 3px rgba(0,0,0,0.5);
  width: inherit;
  padding: 0.65rem 2.25rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-animation-name: eEnBCQ;
  animation-name: eEnBCQ;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  box-shadow: 0 7px 10px rgba(0,0,0,0.5);
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c0:active {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
  box-shadow: 0 0 0 rgba(0,0,0,0.5);
}

<button
  className="c0"
>
  Test
</button>
`;

exports[`Button renders to 1`] = `
.c0 {
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  -webkit-animation-name: cUPaoU;
  animation-name: cUPaoU;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  border: none;
  outline: none;
  background-image: linear-gradient(270deg,#fed29d,#A58B66,#7abae8,#56a0d6);
  background-size: 720%;
  border: none;
  border-radius: 4px;
  box-sizing: border-box;
  font-size: 1.125rem;
  text-align: center;
  color: white;
  font-weight: 400;
  box-shadow: 0 3px 3px rgba(0,0,0,0.5);
  width: inherit;
  padding: 0.65rem 2.25rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-animation-name: eEnBCQ;
  animation-name: eEnBCQ;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  box-shadow: 0 7px 10px rgba(0,0,0,0.5);
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c0:active {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
  box-shadow: 0 0 0 rgba(0,0,0,0.5);
}

<a
  className="c0"
  href="https://ivesvh.com"
  onClick={[Function]}
>
  Test
</a>
`;
